syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.konstfish.bonsai";
option java_outer_classname = "BonsaiProto";
option objc_class_prefix = "HLW";

package bonsai;

// The greeting service definition.
service Server {
  // Sends a greeting
  rpc RegisterClient (RegistrationRequest) returns (RegistrationConfirmation) {}
  rpc PushMetrics (MetricsRequest) returns (MetricsConfirmation) {}
}

// The request message containing the user's name.
message RegistrationRequest {
  string name = 1;
}

// The response message containing the greetings
message RegistrationConfirmation {
  string message = 1;
}


message MetricsRequest {
  string id = 1;
  string job = 2;
  string host = 3;
  bytes metrics = 4;
  map<string, string> labels = 5;
}

message MetricsConfirmation {
  int32 code = 1;
  string confirm = 2;
}